src/state/slices/opportunitiesSlice/opportunitiesSlice.test.ts

```diff
@@ -12,12 +17,54 @@ describe('opportunitiesSlice', () => {
   })
 
   describe('reducers', () => {
+    beforeEach(() => {
+      store.dispatch(opportunities.actions.clear())
+    })
     describe('clear', () => {
       it.todo('clears state back to initial state')
     })
-  })
-
-  describe('selectors', () => {
-    // TODO
+    describe('upsertOpportunityMetadata', () => {
+      it('insert metadata', () => {
+        const payload = {
+          metadata: {
+            'eip155:1:0xMyContract': {
+              // The LP token AssetId
+              assetId: foxEthLpAssetId,
+              provider: DefiProvider.FoxEthLP,
+              tvl: '424242',
+              apy: '0.42',
+              type: DefiType.LiquidityPool,
+              underlyingAssetIds: [foxAssetId, ethAssetId] as [AssetId, AssetId],
+            },
+          },
+          type: 'lp' as const,
+        }
+        const expected = {
+          'eip155:1:0xMyContract': {
+            apy: '0.42',
+            assetId: foxEthLpAssetId,
+            provider: DefiProvider.FoxEthLP,
+            tvl: '424242',
+            type: DefiType.LiquidityPool,
+            underlyingAssetIds: [foxAssetId, ethAssetId],
+          },
+        }
+        store.dispatch(opportunities.actions.upsertOpportunityMetadata(payload))
+        expect(store.getState().opportunities.lp.byId).toEqual(expected)
+      })
+    })
+    describe('upsertUserStakingOpportunities', () => {
+      it('insert user data', () => {
+        const payload = {
+          [serializeUserStakingId('eip155:1/erc20:0xgomes', 'eip155:1:0xMyContract')]: {
```

Comment:
```suggestion
          [serializeUserStakingId('eip155:1/0xgomes', 'eip155:1:0xMyContract')]: {
```

Reply:
u are not an asset ser

Reply:
d4cfcbad0c0e008d2a434b4e87024044a9d45ef8